File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> result = new HashMap();
        for(int i = 0; i < nums.length; i++) {
            if(!result.containsKey(nums[i])) {
                result.put(nums[i],i);
            }
            int complement = target - nums[i];
            if(result.containsKey(complement) && (result.get(complement) != i)){
                return new int[]{i,result.get(complement)};
            }
        }

        return new int[]{};
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        track = {}
        for i in range(len(nums)):
            nTwo = target - nums[i]
            if nTwo in track and track[nTwo] != i:
                return [track[nTwo], i]
            track[nums[i]] = i

        return [-1, -1]